packages/router/src/components/link.rs

```diff
@@ -53,6 +55,21 @@ impl From<&str> for IntoRoutable {
     }
 }
 
+/// Newtype to make is easier to work with rust-url type that other crates commonly use.
+pub struct Url(RustUrl);
```

Comment:
I think since the type IntoRoutable is defined in this crate, can we implement this directly on url::Url (https://doc.rust-lang.org/std/convert/trait.Into.html#implementing-into-for-conversions-to-external-types-in-old-versions-of-rust)

Reply:
Sure, I wasn't sure what the best practice was for working with external types. I thought it was to create a newtype and manipulate that.

Reply:
You beat me to it haha.

Reply:
> Sure, I wasn't sure what the best practice was for working with external types. I thought it was to create a newtype and manipulate that.

I generally only use newtypes to get around rust's rules about foreign traits on foreign items

Reply:
Over my head atm, I will trust your judgment. Thanks for updating it.